lib/Authentication.php

```diff
@@ -468,8 +469,11 @@ public function getEndpoint(): array
      */
     public function setEndpoint(array $endpoint): void
     {
-        if (!array_key_exists('url', $endpoint) || !array_key_exists('region', $endpoint)) {
-            throw new RuntimeException('$endpoint must contain `url` and `region` keys');
+        $hasKey = function(string $key) use ($endpoint){
+            return array_key_exists($key, $endpoint);
+        };
```

Comment:
Could you convert this to an arrow function? Just a little cleaner IMO :)

Reply:
Sure, i can.
but this needs the project to be run with min PHP 7.4.
in composer.json it's "php": ">=7.3".
Can be changed as well, but would cause a major version change for your package, i would assume.

Reply:
Ah good point, my bad. I forgot arrow functions were a >=7.4 feature. Nevermind.